src/sessions.ts

```diff
@@ -441,6 +443,7 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
    * This is here to ensure that ClientSession is never serialized to BSON.
    */
   toBSON(): never {
+    // TODO(NODE-3484): Replace with MongoBSONParseError
```

Comment:
This is one of those unreachables, its not a BSON parse error its really any error just to stop BSON from attempting to serialize a session, fair to leave as a driver error or runtime.

Reply:
The TODO can be removed